pkg/semtech-loramac/contrib/semtech_loramac_radio.c

```diff
@@ -0,0 +1,256 @@
+                       uint8_t hopPeriod, bool iqInverted, uint32_t timeout)
+{
+    (void) fdev;
+    (void) fixLen;
+    sx127x_set_modem(&sx127x, modem);
+    sx127x_set_freq_hop(&sx127x, freqHopOn);
+    sx127x_set_bandwidth(&sx127x, bandwidth);
+    sx127x_set_coding_rate(&sx127x, coding_rate);
+    sx127x_set_spreading_factor(&sx127x, spreading_factor);
+    sx127x_set_crc(&sx127x, crcOn);
+    sx127x_set_freq_hop(&sx127x, freqHopOn);
+    sx127x_set_hop_period(&sx127x, hopPeriod);
+    sx127x_set_fixed_header_len_mode(&sx127x, false);
+    sx127x_set_iq_invert(&sx127x, iqInverted);
+    sx127x_set_payload_length(&sx127x, 0);
+    sx127x_set_tx_power(&sx127x, power);
+    sx127x_set_preamble_length(&sx127x, preambleLen);
+    sx127x_set_rx_single(&sx127x, false);
+    sx127x_set_tx_timeout(&sx127x, timeout * US_PER_MS); /* base unit us, LoRaMAC ms */
+}
+
+uint32_t SX1276GetTimeOnAir(RadioModems_t modem, uint8_t pktLen)
+{
+    (void) modem;
+    return sx127x_get_time_on_air(&sx127x, pktLen);
+}
+
+void SX1276Send(uint8_t *buffer, uint8_t size)
+{
+    netdev_t *dev = (netdev_t *)&sx127x;
+    struct iovec vec[1];
+    vec[0].iov_base = buffer;
+    vec[0].iov_len = size;
+    dev->driver->send(dev, vec, 1);
+}
+
+void SX1276SetSleep(void)
+{
+    sx127x_set_sleep(&sx127x);
+}
+
+void SX1276SetStby(void)
+{
+    sx127x_set_standby(&sx127x);
+}
+
+void SX1276SetRx(uint32_t timeout)
+{
+    (void) timeout;
+    sx127x_set_rx(&sx127x);
+}
+
+void SX1276StartCad(void)
+{
+    sx127x_start_cad(&sx127x);
+}
+
+int16_t SX1276ReadRssi(RadioModems_t modem)
+{
+    sx127x_set_modem(&sx127x, (uint8_t)modem);
+    return sx127x_read_rssi(&sx127x);
+}
+
+void SX1276Write(uint8_t addr, uint8_t data)
+{
+    sx127x_reg_write(&sx127x, addr, data);
+}
+
+uint8_t SX1276Read(uint8_t addr)
+{
+    return sx127x_reg_read(&sx127x, addr);
+}
+
+void SX1276WriteBuffer(uint8_t addr, uint8_t *buffer, uint8_t size)
+{
+    sx127x_reg_write_burst(&sx127x, addr, buffer, size);
+}
+
+void SX1276ReadBuffer(uint8_t addr, uint8_t *buffer, uint8_t size)
+{
+    sx127x_reg_read_burst(&sx127x, addr, buffer, size);
+}
+
+void SX1276SetMaxPayloadLength(RadioModems_t modem, uint8_t max)
+{
+    (void) modem;
+    sx127x_set_max_payload_len(&sx127x, max);
+}
+
+bool SX1276CheckRfFrequency(uint32_t frequency)
+{
+    (void) frequency;
+    /* Implement check. Currently all frequencies are supported */
+    return true;
+}
+
+void SX1276SetTxContinuousWave(uint32_t freq, int8_t power, uint16_t time)
+{
+    (void) freq;
+    (void) power;
+    (void) time;
+    /* TODO */
+}
+
+void SX1276SetPublicNetwork(bool enable)
+{
+    if (enable) {
+        /* Use public network syncword */
+        sx127x_set_syncword(&sx127x, LORA_SYNCWORD_PUBLIC);
+    }
+    else {
+        /* Use private network syncword */
+        sx127x_set_syncword(&sx127x, LORA_SYNCWORD_PRIVATE);
+    }
+}
+
+/**
+ * LoRa function callbacks
+ */
+const struct Radio_s Radio =
```

Comment:
Aaaaalright then I agree it would be better to rename with a generic name -> sx127x.
As I recall (correct me if I'm wrong), the Radio_s struct is define for SX1272 and SX1276 (In Semtech code) and as @jia200x said this is an abstraction layer. This level of abstraction exists because Semtech did not write generic functions from the beginning so function like SX1272Init and SX1276Init are indeed different right ? (because SX1272 and SX1276 are different chips)
In our case, we already wrote generic functions from the beginning so we only need the Radio_s struct to match Semtech application code. 
I would prefer to rename directly all theses functions to RIOT current API instead of keeping SX1276xxxx stuff as it's very confusing...

Reply:
Just pushed the update

Reply:
> As I recall (correct me if I'm wrong), the Radio_s struct is define for SX1272 and SX1276 (In Semtech code) and as @jia200x said this is an abstraction layer. This level of abstraction exists because Semtech did not write generic functions from the beginning so function like SX1272Init and SX1276Init are indeed different right ? (because SX1272 and SX1276 are different chips)

Exactly :)